Sources/Overlays/_Testing_Foundation/Attachments/Attachment+URL.swift

```diff
@@ -83,79 +81,177 @@ extension Attachment where AttachableValue == Data {
+        if #available(_uttypesAPI, *) {
+          return (preferredName as NSString).appendingPathExtension(for: .zip)
         }
 #endif
-        return (preferredName as NSString).appendingPathExtension("tgz") ?? preferredName
+        return (preferredName as NSString).appendingPathExtension("zip") ?? preferredName
       }()
+    }
+
+#if SWT_TARGET_OS_APPLE
+    let data: Data = try await withCheckedThrowingContinuation { continuation in
+      let fileCoordinator = NSFileCoordinator()
+      let fileAccessIntent = NSFileAccessIntent.readingIntent(with: url, options: [.forUploading])
 
-      try await self.init(Data(compressedContentsOfDirectoryAt: url), named: preferredName, sourceLocation: sourceLocation)
+      fileCoordinator.coordinate(with: [fileAccessIntent], queue: Self._operationQueue) { error in
+        let result = Result {
+          if let error {
+            throw error
+          }
+          return try Data(contentsOf: fileAccessIntent.url, options: [.mappedIfSafe])
+        }
+        continuation.resume(with: result)
+      }
+    }
+#else
+    let data = if isDirectory {
+      try await compressContentsOfDirectory(at: url)
     } else {
       // Load the file.
-      try self.init(Data(contentsOf: url, options: [.mappedIfSafe]), named: preferredName, sourceLocation: sourceLocation)
+      try Data(contentsOf: url, options: [.mappedIfSafe])
     }
+#endif
+
+    self.init(data, named: preferredName, sourceLocation: sourceLocation)
   }
 }
 
-// MARK: - Attaching directories
+#if !SWT_NO_PROCESS_SPAWNING && os(Windows)
+/// The filename of the archiver tool.
+private let _archiverName = "tar.exe"
 
-extension Data {
-  /// Initialize an instance of this type by compressing the contents of a
-  /// directory.
-  ///
-  /// - Parameters:
-  ///   - directoryURL: A URL referring to the directory to attach.
-  ///
-  /// - Throws: Any error encountered trying to compress the directory, or if
-  ///   directories cannot be compressed on this platform.
-  ///
-  /// This initializer asynchronously compresses the contents of `directoryURL`
-  /// into an archive (currently of `.tgz` format, although this is subject to
-  /// change) and stores a mapped copy of that archive.
-  init(compressedContentsOfDirectoryAt directoryURL: URL) async throws {
-    let temporaryName = "\(UUID().uuidString).tgz"
-    let temporaryURL = FileManager.default.temporaryDirectory.appendingPathComponent(temporaryName)
+/// The path to the archiver tool.
+///
+/// This path refers to a file (named `_archiverName`) within the `"System32"`
+/// folder of the current system, which is not always located in `"C:\Windows."`
+///
+/// If the path cannot be determined, the value of this property is `nil`.
+private let _archiverPath: String? = {
+  let bufferCount = GetSystemDirectoryW(nil, 0)
+  guard bufferCount > 0 else {
+    return nil
+  }
+
+  return withUnsafeTemporaryAllocation(of: wchar_t.self, capacity: Int(bufferCount)) { buffer -> String? in
```

Comment:
```suggestion
  return withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(bufferCount)) { buffer -> String? in
```

Reply:
I pretty consistently use `wchar_t` in the Windows code here, but we can consider changing to `WCHAR` in a later PR if you think it's important.